Documentation/Maintainers/Codesnippets.rst

```diff
@@ -0,0 +1,60 @@
+..  include:: /Includes.rst.txt
+..  _codesnippet-generation:
+
+=======================
+Code snippet generation
+=======================
+
+..  todo: Document how to create additional code snippets
+
+..  _codesnippet-regeneration:
+
+Regenerate existing code snippets
+=================================
+
+To regenerate the existing code snippets in a manual or extension clone
+the repository of that manual.
+
+..  code-block:: bash
+
+    git clone git@github.com:TYPO3-Documentation/TYPO3CMS-Reference-CoreApi.git
+
+
+Use composer to install all dependencies,
+including dev-dependencies. In official manuals we have a make command for
+your convenience.
+
+..  tabs::
+
+    ..  tab:: Make
+
+        ..  code-block:: bash
+
+            make install
+
+    ..  tab:: composer
+
+        ..  code-block:: bash
+
+            # in repositories where the composer.lock is not in version control
```

Comment:
Wouldn't "composer install" be the way here? 

Where composer.lock exists, this would be respected and install locked versions.
Where composer.lock does not exist, the install would create it.

Using "composer update" would override composer.lock versions in this case, which could be problematic?!

Reply:
Actually when you don't have the composer.json in VCS you want to always do composer update. As you still have a local composer.lock. Otherwise you will not get the newest version like someone who makes a fresh install would have

Reply:
The HowTo-text starts with cloning a repo - how should a local composer.lock exist if there is none in repo?

Reply:
Its a common mistake that we make if we already cloned the repository and installed it before. To do composer install and not composer update and then you will have outdated codesnippets.

Reply:
ok, I understand - but then I would suggest to mention that directly

and what about repositories where composer.lock is in VCS? As you addressed the case that it is not in VCS we should also mention the other case, don´t we?

```suggestion
            # in repositories with composer.lock in version control
            composer install
            
            # in repositories where the composer.lock is not in version control
            # to prevent outdated packages due to a local composer.lock from earlier install-runs 
```